### ISSUE TITLE
Improve method of asserting that specific errors occur during tests.

### ISSUE THREAD

**Issue by tjprescott** [Azure/azure-cli#5155]
With VCRTestBase we had the ability to allow commands to succeed if an expected error was encountered.

The current scenario test allows for "expect_failure" but it will allow the command to succeed if ANY error is encountered.

The mechanism of using:
```Python
with self.assertRaises(CLIError):
   self.cmd(...)
```

Doesn't work consistently and is requiring changing CLIError to AssertRaises. This may be an unintended side effect of the Knack conversion.

Currently we are working around this, but we should develop a comprehensive solution and update these cases.

**Comment by tjprescott** [comment]
Specifically, you should be able to assert that a specific expected error occurred by examining the error message. Simply looking for "CLIError" is insufficient (and doesn't even work reliably currently).  We often have workarounds because of a poor service error message. Detecting when that error message is fixed would fail the test, alert us, and allow us to remove the workaround. Also, you want to ensure the command fails for the reason you think it should.

**Comment by derekbekoe** [comment]
Is this what this change is for?
https://github.com/Microsoft/knack/pull/66

**Comment by tjprescott** [comment]
Kinda. I think that fixes #5032. The issue here is more an issue with ScenarioTest being too coarse-grained when it comes to validating the error.

**Comment by tjprescott** [comment]
```Python
        with self.assertRaisesRegexp(CLIError, "usage error: --vnet NAME --subnet NAME | --vnet ID --subnet NAME | --subnet ID"):
            self.cmd('container create -g {rg} -n {container_group_name} --image nginx --vnet {vnet_name}')
```

This syntax is both concise and powerful for asserting specific errors.
